### PR TITLE
PercentChangePostprocessor feature

### PR DESCRIPTION
**Pull request by mangerij**
User can use this to calculate the percent change of a postprocessor between time steps

Reference to https://groups.google.com/forum/#!topic/moose-users/6b3ZGLxhW5g

and Issue #5297

### CODE REVIEW

**Comment by friedmud** [comment]
Firstly, thanks for putting this together!  We just need to do a few things to fix this up before we can merge it:

1.  The test failed because your commit message needs to reference your ticket (ie, your commit message must have #5297 in it somewhere).  We'll fix this up when we fix the next thing...
1.  No need for 3 commits for this small feature.  We need to "squash" those three commits into 1.  Here's how you do that:
  - `git rebase -i HEAD~4` That will give you a screen where you can edit your commit history (for the last 4 commits) on this branch.  (Note: this is typically in vi at this point... so either look up vi commands or set your preferred editor using `git config --global core.editor "editor"` before issuing this command)
  - What you want to do here is put an `s` (for "squash") in front of the second and third commits to squash them into the "top" one.  Then save and exit your editor.  Refer to this for help: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html
  - Then it will open an editor where you can change your commit message.  Delete all of the previous messages and put a new one in that just says `Adding PercentChangePostprocessor closes #5297`.  Then save and close the file.
1.  Now do a `git log` and you should see your _one_ new commit at the top of the stack.
1.  Now push that back to github on this branch.  You'll need to use `--force` to push the branch because you changed history.

Let me know if you have any trouble.

**Comment by friedmud** [comment]
@mangerij You don't need to close this PR or anything.  Just follow my instructions and force push to this branch again and the PR will automatically update.

**Comment by mangerij** [comment]
Alright cool!

Sorry. I'm a git n00b

**Comment by friedmud** [comment]
No problem at all!  We were all there once.  Take your time and work through this.  Let us know if you have any questions.

**Comment by permcody** [comment]
Invalidated due to merge of #5285

**Comment by permcody** [comment]
@mangerij - Thanks for the contribution. I have a few style comments that I'm going to make before I merge this that I hope you'll be kind enough to fix for us. For your reference you can find our CodeStandards here: http://mooseframework.org/wiki/CodeStandards/

File: framework/include/postprocessors/PercentChangePostprocessor.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ class PercentChangePostprocessor : public GeneralPostprocessor
+ {
+ public:
+   PercentChangePostprocessor(const std::string & name, InputParameters parameters);
+   virtual ~PercentChangePostprocessor();
+   virtual void initialize();
+   virtual void execute();
+   virtual Real getValue();
+ protected:
+   const PostprocessorValue &_postprocessor,&_postprocessor_old;
```
Comment (permcody) [inline framework/include/postprocessors/PercentChangePostprocessor.h]: Always put space around operators and symbols like "&" and "*".

File: framework/src/postprocessors/PercentChangePostprocessor.C [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+PercentChangePostprocessor::PercentChangePostprocessor(const std::string & name, InputParameters parameters) :
+    GeneralPostprocessor(name, parameters),
+    _postprocessor(getPostprocessorValue("postprocessor")),
+    _postprocessor_old(getPostprocessorValueOld("postprocessor"))
+
+{
+}
+
+PercentChangePostprocessor::~PercentChangePostprocessor(){
```
Comment (permcody) [inline framework/src/postprocessors/PercentChangePostprocessor.C]: Curly braces should always be placed on their own line. Opening and closing blocks should be aligned.

File: framework/src/postprocessors/PercentChangePostprocessor.C [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+void
+PercentChangePostprocessor::execute(){
+}
+
+Real
+PercentChangePostprocessor::getValue()
+{
+  return fabs( ( fabs(_postprocessor)- fabs(_postprocessor_old) )*pow(fabs(_postprocessor),-1));
```
Comment (permcody) [inline framework/src/postprocessors/PercentChangePostprocessor.C]: - When using a method from the standard library always make sure you prefix it with "std::"
- Don't put a space after an open parentheses
- Put space around the "*" operator and after the comma

**Comment by permcody** [comment]
@mangerij - Great, thanks for fixing that all up. Since this has turned into a full PR tutorial, I might as well subject you to the full cycle...

The last thing you should do here is squash these commits into one. Generally we like to keep commits that are only meaningful or adding new capability, fixing whitespace or style is best squashed so we only see the real diffs in the history. Just do an interactive rebase against the devel branch "git rebase -i <branch>" and put an "s" next to the second commit. Finally do a force push to replace these two commits with your one new one. 

Thanks for being patient with us and adding your name to our contributors list!

**Comment by permcody** [comment]
Squashed and repushed while maintaining authorship in #5300. Thanks John!

**Pull request closed without merging**